### DISCUSSION TITLE
[Enhancement] API Action to use Base URL with Paths

### DISCUSSION THREAD

**Discussion by VPCU** [danny-avila/LibreChat#2811] [Feature Requests & Suggestions]
### What happened?

Assistants fail to run actions because the API calls use wrong base URL.

According to OpenAPI 3.0 documents,

> All API endpoints are relative to the base URL. For example, assuming the base URL of https://api.example.com/v1, the /users endpoint refers to https://api.example.com/v1/users.
> In OpenAPI 3.0, you use the servers array to specify one or more base URLs for your API.

### Steps to Reproduce

In assistant builder, add an action with the schema:
```
openapi: 3.0.1
servers:
  - url: https://example.com/a/b/c
paths:
  /d:
...
```

The action would throw an error like "Request failed with status code 404".

However the following schema works:

```
openapi: 3.0.1
servers:
  - url: https://example.com
paths:
  /a/b/c/d:
...
```

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
debug: [required actions] user: <> | thread_id: <>| run_id: <>
["{\"tool\":\"<>\",\"toolInput\":<>... [truncated]"]

error: Request failed with status code 404
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
According to my research, both ways are technically correct in OpenAPI 3.0. I was mainly following suit with ChatGPT with this feature, as this behavior can be observed with their custom GPTs as well.

While supporting both ways would be a QOL improvement, as some server setups could benefit from a less redundant spec including paths repeatedly, I've properly marked this as a suggestion and not a bug.
